packages/react/src/components/Table/Table.main.story.jsx

```diff
@@ -1133,6 +1134,106 @@ WithFiltering.parameters = {
   },
 };
 
+export const WithCustomInputFiltering = () => {
+  const {
+    selectedTableType,
+    hideClearAllFiltersButton,
+    hasEmptyFilterOption,
+    hasMultiSelectFilter,
+    hasFilterRowIcon,
+  } = getTableKnobs({
+    knobsToCreate: [
+      'selectedTableType',
+      'hideClearAllFiltersButton',
+      'hasEmptyFilterOption',
+      'hasMultiSelectFilter',
+      'hasFilterRowIcon',
+    ],
+    getDefaultValue: (knobName) => {
+      if (knobName === 'hideClearAllFiltersButton') {
+        return false;
+      }
+
+      if (knobName === 'hasEmptyFilterOption') {
+        return false;
+      }
+
+      if (knobName === 'hasMultiSelectFilter') {
+        return false;
+      }
+
+      if (knobName === 'hasFilterRowIcon') {
+        return false;
+      }
+
+      return true;
+    },
+  });
+
+  const MyTable = selectedTableType === 'StatefulTable' ? StatefulTable : Table;
+  const data = hasEmptyFilterOption
+    ? getTableDataWithEmptySelectFilter().slice(0, 30)
+    : getTableData().slice(0, 30);
+
+  const columns = decorateTableColumns(
+    getTableCustomColumns(),
+    hasEmptyFilterOption,
+    hasMultiSelectFilter
+  ).map((col) =>
+    col.id === 'object'
+      ? {
+          ...col,
+          tooltip: `This column has objects as values and needs a custom filter function that
+      filters based on an object property.`,
+        }
+      : col
+  );
+
+  const activeFilters = object('Active filters (view.filters)', [
+    {
+      columnId: 'string',
+      value: 'whiteboard',
+    },
+  ]);
+  const activeBar = select(
+    'Show filter toolbar (view.toolbar.activeBar)',
+    ['filter', undefined],
+    'filter'
+  );
+
+  const filteredData = data;
```

Comment:
better if you  remove this line and use the `data` variable in line 1212 directly